buildHooks/src/comparePluginOverrides.ts

```diff
@@ -0,0 +1,393 @@
+                                    '--ignore-space-change',
+                                    ourPath,
+                                    incPath,
+                                ],
+                                { reject: false }
+                            ).exitCode === 0
+                    )
+                );
+            return {
+                pluginName,
+                incomingVersion,
+                paths,
+                ourPluginPath,
+                incomingPluginPath,
+                ourUsedVersion: getContext().files.scopedConfigTemplates['rnv']?.pluginTemplates?.[pluginName]?.version,
+                ...(!difference
+                    ? { kind: 'matching' }
+                    : paths.some(({ ourPath }) => fs.existsSync(ourPath))
+                    ? { kind: 'different', difference }
+                    : { kind: 'new', difference, matchesOurVersions: findMatchingVersions() }),
+            };
+        }) ?? []
+    );
+};
+
+const getOverrideSummary = (ov: Override) => {
+    let headline = chalks[ov.kind](`${ov.pluginName}${ov.version ? `@${ov.version}` : ''} ${ov.kind}`);
+    if (ov.kind === 'new' && ov.matchesOurVersions?.length) {
+        headline = `${headline}, but same as our ${ov.matchesOurVersions
+            .map((ourVersion) => (ourVersion ? `@${ourVersion}` : 'versionless'))
+            .join(', ')}`;
+    }
+    const difference = ov.kind === 'matching' ? undefined : ov.difference;
+    return [
+        headline,
+        difference?.diffStat,
+        difference?.incGitLog.length && formatGitLog(difference.incGitLog),
+        ov.kind !== 'matching' &&
+            chalk().gray(
+                ov.ourUsedVersion ? `We're using version ${ov.ourUsedVersion}` : 'We seem to not be using this'
+            ),
+        difference?.ourGitLog.length && formatGitLog(difference.ourGitLog),
+    ]
+        .filter((i) => i)
+        .join('\n');
+};
+
+const showOverrideDiff = (ov: Override) => {
+    if (ov.kind === 'matching') return;
+    paginate(`${getOverrideSummary(ov)}\n\n${ov.difference.diff ?? ''}`);
+};
+
+const applyOverridePatch = (ov: Override) => {
+    if (ov.kind === 'matching') return;
+    fs.mkdirSync(ov.ourPluginPath, { recursive: true });
+    const tempPath = path.join(fs.mkdtempSync(path.join(os.tmpdir(), 'rnv-plugins-script-')), 'plugin.patch');
+    fs.writeFileSync(tempPath, ov.difference.patch);
+    execa.sync(
+        'patch',
+        [
+            '--version-control',
+            'none',
+            '--strip',
+            '1',
+            '--remove-empty-files',
+            '--ignore-whitespace',
+            '--input',
+            tempPath,
+        ],
+        {
+            cwd: ov.ourPluginPath,
+        }
+    );
+    fs.unlinkSync(tempPath);
+};
+
+const promptOverride = async (ov: Override) => {
+    if (ov.kind === 'matching') return;
+    console.clear();
+    console.log(getOverrideSummary(ov));
+    showOverrideDiff(ov);
+    console.log();
```

Comment:
are console logs needed here?

Reply:
Yeah, to still show the summary when the pager is closed.